packages/babel-core/src/transformation/internal-plugins/shadow-functions.js

```diff
@@ -48,8 +48,16 @@ function remap(path, key) {
 
   let currentFunction;
   let passedShadowFunction = false;
+  let first = true;
 
-  let fnPath = path.findParent(function (path) {
+  let fnPath = path.find(function (path) {
+    if (path.parentPath && path.parentPath.isClassProperty() && path.key === "value") {
+      return true;
+    }
+    if (first) {
```

Comment:
Rather than using this flag, which took me a minute to process, could we perhaps do

```
let fnPath = path.find(function (innerPath) { // rename argument
    if (innerPath.parentPath && innerPath.parentPath.isClassProperty() && innerPath.key === "value") {
        return true;
    }
    if (path === innerPath) return false;
```